src/bin/lychee/main.rs

```diff
@@ -184,6 +231,68 @@ async fn run(cfg: &Config, inputs: Vec<Input>) -> Result<i32> {
     }
 }
 
+async fn recurse(
+    response: Response,
+    cache: &mut Cache,
+    input_domains: &HashSet<String>,
+    cfg: &Config,
+    pb: &Option<ProgressBar>,
+    send_req: Sender<Request>,
+) -> Result<usize> {
+    let recursion_level = response.recursion_level + 1;
+
+    if let Some(depth) = cfg.depth {
+        if recursion_level > depth {
+            // Maximum recursion depth reached; stop link checking.
+            return Ok(0);
+        }
+    }
+
+    if !response.status.is_success() {
+        return Ok(0);
+    }
+    if cache.contains(response.uri.as_str()) {
```

Comment:
Nitpick: I'm a bit skeptical of the name `cache` here. We're not really caching, we're merely "checking if this URI was checked before, and should therefore be skipped".

Reply:
How about `seen`?